src/docs/howto_develop.md

```diff
@@ -167,6 +167,13 @@ To run tests for a specific target, run the below, substituting the target(s) wi
     :::bash
     $ ./pants test tests/python/pants_test/util:strutil
 
+Before running integration tests, you must first bootstrap `pants.pex` as below. Note that this
```

Comment:
So, while I'm fine with requiring the pex to run integration tests (it makes it the cost of integration tests more apparent, yay!), I don't think that it is reasonable to require people to remember to run this.

I think that you'll need to design a solution to either: 1) trigger the pex (re-)build automatically, 2) fail if it hasn't been rebuilt. I think that it is fine for it to be (relatively) hacky, but it's complicated by the fact that this needs to be consumed by both v1 and v2. Given that it need to support both v1 and v2, a few potential locations to insert the logic:
1. in the `pants` script itself
2. as a side-effect in `register.py` for a repo-private backend (https://github.com/pantsbuild/pants/tree/master/pants-plugins)
3. via some other not-yet-existing "load hook" functionality

In all cases there is the additional challenge of identifying that we are running integration tests though.

The advantage of the `register.py` sideeffect might be that

Reply:
Totally reasonable to ask for this first. Will think more about this this week and hopefully come up with a solution soon. @cosmicexplorer mentioned over Slack the invalidation code we use for re-bootstrapping `native_engine.so` as potential inspiration. 

Thanks for these suggestions!

Reply:
The scheme to maintain a cache of `pants.pex`, as described in https://github.com/pantsbuild/pants/issues/8209, is now complete: https://github.com/pantsbuild/pants/pull/8183/files#diff-b5bfb910b73c1c80b6144d21ce1ea199

The main remaining task is to figure out how to automatically call `bootstrap_pants_pex.sh`. The best I can think of is to check in `pants` if any of the CLI args start with `test`? This would mean any test, including any unit tests, would call the script to ensure it has an up-to-date `pants.pex`. A better scheme would differentiate between integration and unit tests, but I don't know how to do this.

Reply:
This is a wild thought:
Could the `OptionsBootstrapper` be modified to know what goal(s) it's trying to run? This is the place where we "parse the bare minimum we need of the command line".

That way, we could write a very concise `main()` function, that would parse the cli, and check the goals. My main goal for this is to make sure that we're avoiding ad-hoc command line parsing, and localize it as much as possible.

Another options is to formalize [this functionality](https://github.com/pantsbuild/pants/blob/master/src/python/pants/bin/pants_runner.py#L31) under a class, and wrap that in a script. This is related to pantsd, but it has some goal parsing involved.

As for differentiating between unit/integration test, unfortunately `--tag` is not a bootstrap option, so this solution wouldn't be able to tell them apart.